src/main/java/com/hivemq/persistence/util/AbstractFutureUtils.java

```diff
@@ -34,167 +36,87 @@
+    private static final AnyToVoidFunction ANY_TO_VOID_FUNCTION = new AnyToVoidFunction();
 
+    public abstract @NotNull ListenableFuture<Void> voidFutureFromAnyFuture(@NotNull ListenableFuture<?> anyFuture);
 
-    public abstract ListenableFuture<Void> voidFutureFromAnyFuture(final ListenableFuture<?> anyFuture);
-
-
-    public ListenableFuture<Void> voidFutureFromAnyFuture(final ListenableFuture<?> anyFuture, final ExecutorService executorService) {
-        return Futures.transformAsync(anyFuture, new AnyToVoidAsyncFunction(), executorService);
+    public @NotNull ListenableFuture<Void> voidFutureFromAnyFuture(
+            final @NotNull ListenableFuture<?> anyFuture, final @NotNull Executor executor) {
+        return Futures.transform(anyFuture, ANY_TO_VOID_FUNCTION, executor);
     }
 
-    public ListenableFuture<Void> mergeVoidFutures(final ListenableFuture<Void> future1, final ListenableFuture<Void> future2) {
+    public @NotNull ListenableFuture<Void> mergeVoidFutures(
+            final @NotNull ListenableFuture<Void> future1, final @NotNull ListenableFuture<Void> future2) {
         return voidFutureFromList(ImmutableList.of(future1, future2));
     }
 
-    public ListenableFuture<Void> voidFutureFromList(final ImmutableList<ListenableFuture<Void>> futureList) {
-
-        if (futureList.isEmpty()) {
-            return Futures.immediateFuture(null);
-        }
-
-        final SettableFuture<Void> resultFuture = SettableFuture.create();
-        final FutureCallback<Void> futureCallback = new VoidFutureCombiningCallback(futureList.size(), resultFuture);
-        for (final ListenableFuture<Void> voidListenableFuture : futureList) {
-
-            Futures.addCallback(voidListenableFuture, futureCallback, MoreExecutors.directExecutor());
-        }
-
-        return resultFuture;
-    }
-
-    public ListenableFuture<Void> voidFutureFromAnyFutureList(final ImmutableList<ListenableFuture> futureList) {
+    public <T> @NotNull ListenableFuture<Void> voidFutureFromList(final @NotNull ImmutableList<ListenableFuture<T>> futureList) {
 
         if (futureList.isEmpty()) {
             return Futures.immediateFuture(null);
         }
 
         final SettableFuture<Void> resultFuture = SettableFuture.create();
-        final FutureCallback<Void> futureCallback = new VoidFutureFromAnyCombiningCallback(futureList.size(), resultFuture);
-        for (final ListenableFuture voidListenableFuture : futureList) {
+        final FutureCallback<T> futureCallback = new VoidFutureCombiningCallback<>(futureList.size(), resultFuture);
 
+        for (final ListenableFuture<T> voidListenableFuture : futureList) {
             Futures.addCallback(voidListenableFuture, futureCallback, MoreExecutors.directExecutor());
         }
 
         return resultFuture;
     }
 
-    public <T> void addSettableFutureCallback(final ListenableFuture<T> listenableFuture, final SettableFuture<T> settableFuture) {
+    public <T> void addSettableFutureCallback(
```

Comment:
All method calls to this method can be replaced with `settableFuture.setFuture(listenableFuture)`
This method should then be removed

Reply:
The method wasn't even used. I have removed it.